src/ripple/app/ledger/impl/InboundLedger.cpp

```diff
@@ -898,33 +898,40 @@ InboundLedger::takeTxNode(
         return true;
     }
 
-    auto nodeIDit = nodeIDs.cbegin();
-    auto nodeDatait = data.begin();
-    TransactionStateSF filter(
-        mLedger->txMap().family().db(), app_.getLedgerMaster());
-
-    while (nodeIDit != nodeIDs.cend())
+    try
     {
-        if (nodeIDit->isRoot())
-        {
-            san += mLedger->txMap().addRootNode(
-                SHAMapHash{mLedger->info().txHash},
-                makeSlice(*nodeDatait),
-                snfWIRE,
-                &filter);
-            if (!san.isGood())
-                return false;
-        }
-        else
+        auto nodeIDit = nodeIDs.cbegin();
+        auto nodeDatait = data.begin();
+        TransactionStateSF filter(
+            mLedger->txMap().family().db(), app_.getLedgerMaster());
+
+        while (nodeIDit != nodeIDs.cend())
         {
-            san += mLedger->txMap().addKnownNode(
-                *nodeIDit, makeSlice(*nodeDatait), &filter);
-            if (!san.isGood())
-                return false;
-        }
+            if (nodeIDit->isRoot())
+            {
+                san += mLedger->txMap().addRootNode(
+                    SHAMapHash{mLedger->info().txHash},
+                    makeSlice(*nodeDatait),
+                    &filter);
+                if (!san.isGood())
+                    return false;
+            }
+            else
+            {
+                san += mLedger->txMap().addKnownNode(
+                    *nodeIDit, makeSlice(*nodeDatait), &filter);
+                if (!san.isGood())
+                    return false;
+            }
 
-        ++nodeIDit;
-        ++nodeDatait;
+            ++nodeIDit;
+            ++nodeDatait;
+        }
+    }
+    catch (std::exception const& ex)
+    {
+        JLOG(m_journal.error()) << "Peer sent bad tx node data: " << ex.what();
```

Comment:
OCD nit: consider either `Peer sent bad **TX** node data:` or `Peer sent bad transaction node data:`

Reply:
`san.incInvalid()` should probably be called here.